spark/v3.1/spark/src/main/java/com/netease/arctic/spark/writer/KeyedSparkBatchWrite.java

```diff
@@ -261,6 +261,33 @@ private abstract static class AbstractWriterFactory implements DataWriterFactory
       this.transactionId = transactionId;
       this.orderedWrite = orderedWrite;
     }
+
+    public TaskWriter<InternalRow> newDeleteWriter(int partitionId, long taskId, StructType schema) {
+      return TaskWriters.of(table)
+          .withTransactionId(transactionId)
+          .withPartitionId(partitionId)
+          .withTaskId(taskId)
+          .withDataSourceSchema(schema)
+          .newChangeWriter();
+    }
+
+    public TaskWriter<InternalRow> newInsertWriter(int partitionId, long taskId, StructType schema) {
+      if (orderedWrite) {
+        return TaskWriters.of(table)
+            .withTransactionId(transactionId)
+            .withPartitionId(partitionId)
+            .withTaskId(taskId)
+            .withDataSourceSchema(schema)
+            .withOrderedWriter(true)
+            .newChangeWriter();
+      }
+      return TaskWriters.of(table)
+          .withTransactionId(transactionId)
+          .withPartitionId(partitionId)
+          .withTaskId(taskId)
+          .withDataSourceSchema(schema)
+          .newChangeWriter();
```

Comment:
What's diferent between those two method?

the args of orderedwrite cloud simplied.